src/modules/job-manager/priority.c

```diff
@@ -97,7 +97,8 @@ void priority_handle_request (flux_t *h,
     orig_priority = job->priority;
     if (event_job_post_pack (ctx->event, job,
                              "priority",
-                             "{ s:i s:i }",
+                             "{ s:I s:i s:i }",
+                             "id", job->id,
                              "userid", cred.userid,
                              "priority", priority) < 0)
         goto error;
```

Comment:
We're going to need every event in the events RPC stream to carry a job ID.

If we change every `event_job_post_pack()` call, then we'll be including the job ID in the KVS where it's already known (since it's one eventlog per job there).

Could we look for a place to add this to all events, and only for the RPC events stream?

One idea would be to enclose each eventlog entry in an outer object, just for the RPC stream, e.g.
```json
{
   "id":1234,
    "event":{
      "timestamp":42.24,
      "name":"start"
    }
}
```

That way the original event object can remain unchanged, and continue to be efficiently `json_incref()`ed for the RPC stream.

We also wouldn't need to update all the event definitions in RFC 21.